org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/ui/util/TargetPlatformUtil.java

```diff
@@ -46,6 +46,7 @@ public static void setTargetPlatform() throws Exception {
 	 * @since 2.14
 	 */
 	public static void setTargetPlatform(Class<?> context) throws Exception {
+		if (Boolean.getBoolean("eclipse.pde.launch")) return;
```

Comment:
Nitpick: I'd extract a method, e.g. isPdeLaunch() and also do a linebreak before the return. But in general this looks about right to me.